components/mdns/mdns.c

```diff
@@ -6435,7 +6549,14 @@ esp_err_t mdns_service_subtype_update_multiple_items_for_host(const char *instan
     mdns_srv_item_t *s = _mdns_get_service_item_instance(instance_name, service_type, proto, hostname);
     ESP_GOTO_ON_FALSE(s, ESP_ERR_NOT_FOUND, err, TAG, "Service doesn't exist");
 
-    // TODO: find subtype needs to say sendbye
+    mdns_subtype_t *goodbye_subtype = NULL;
+    _mdns_service_find_subtype_needed_sendbye(s->service, subtype, num_items, &goodbye_subtype);
```

Comment:
would it be easier to return the list directly?
```suggestion
    mdns_subtype_t *goodbye_subtype = _mdns_service_find_subtype_needed_sendbye(s->service, subtype, num_items);
```